core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/useragent/SdkUserAgentBuilder.java

```diff
@@ -124,4 +134,12 @@ private static void appendAdditionalJvmMetadata(StringBuilder builder, SystemUse
             appendNonEmptyField(builder, METADATA, lang);
         }
     }
+
+    private static void checkLengthAndWarn(String appId) {
+        if (appId.length() > 50) {
```

Comment:
Just to double check, is this only logged once?

Reply:
Yes, it's only logged once.